### PR TITLE
gh-86457: Add docs for Argument Clinic @text_signature directive

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-86457

----
:books: Documentation preview :books:: https://cpython-previews--107747.org.readthedocs.build/

### CODE REVIEW

File: Doc/howto/clinic.rst [hunk 1]
```diff
@@ -1900,6 +1900,39 @@ blocks embedded in Python files look slightly different.  They look like this:
     #/*[python checksum:...]*/
 
 
+.. _clinic-howto-override-signature:
+
+How to override the generated signature
+---------------------------------------
+
+You can use the ``@text_signature`` directive to override the generated
```
Comment (AlexWaygood) [inline Doc/howto/clinic.rst]:
```suggestion
You can use the ``@text_signature`` directive to override the default generated
```

File: Doc/howto/clinic.rst [hunk 2]
```diff
@@ -1900,6 +1900,39 @@ blocks embedded in Python files look slightly different.  They look like this:
...
+signature in the docstring.
+This can be useful for complex signatures that Argument Clinic cannot handle.
+The ``@text_signature`` directive takes one argument;
```
Comment (AlexWaygood) [inline Doc/howto/clinic.rst]:
```suggestion
The ``@text_signature`` directive takes one argument:
```

File: Doc/howto/clinic.rst [hunk 3]
```diff
@@ -1900,6 +1900,39 @@ blocks embedded in Python files look slightly different.  They look like this:
...
+the custom signature as a string.
+The provided signature is copied verbatim to the generated docstring.
+Example from :source:`Objects/codeobject.c`::
```
Comment (AlexWaygood) [inline Doc/howto/clinic.rst:1915]:
```suggestion

Example from :source:`Objects/codeobject.c`::
```

File: Doc/howto/clinic.rst [hunk 4]
```diff
@@ -1900,6 +1900,39 @@ blocks embedded in Python files look slightly different.  They look like this:
...
+   code.replace
+       *
+       co_argcount: int(c_default="self->co_argcount") = unchanged
+       co_posonlyargcount: int(c_default="self->co_posonlyargcount") = unchanged
+       # etc ...
+
+       Return a copy of the code object with new values for the specified fields.
+   [clinic start generated output]*/
+
+The generated docstring ends up like this::
```
Comment (AlexWaygood) [inline Doc/howto/clinic.rst]:
```suggestion
The generated docstring ends up looking like this::
```

**Review by AlexWaygood — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Thanks for the review!

**Comment by erlend-aasland** [comment]
I will wait with the backports until #107744 and #107746 have landed.

**Comment by picnixz** [comment]
No need for a 3.12 backports since AC is now documented in the devguide.

**Pull request merged**